### PR TITLE
[PM-11477] Remove deprecated cryptoservice functions

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11477
https://bitwarden.atlassian.net/browse/PM-11475

## 📔 Objective

Before moving cryptoservice to key-management ownership and renaming to "keyservice" this PR removes the deprecated functions that belong in encrypt service, and migrates the uses to the keyservice functions.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/common/src/tools/send/models/domain/send.ts [hunk 1]
```diff
@@ -73,9 +73,10 @@ export class Send extends Domain {
     const model = new SendView(this);
 
     const cryptoService = Utils.getContainerService().getCryptoService();
+    const encryptService = Utils.getContainerService().getEncryptService();
 
     try {
-      model.key = await cryptoService.decryptToBytes(this.key, null);
+      model.key = await encryptService.decryptToBytes(this.key, null);
       model.cryptoKey = await cryptoService.makeSendKey(model.key);
```
[the remark this answers is not in the document]
Comment (quexten) [reply libs/common/src/tools/send/models/domain/send.ts:80]: I don't think makesendkey will be part of EncryptService. It seems more like a cryptoservice (in the future "keyservice") function. cc @justindbaur @jlf0dev correct me if I'm wrong.

File: apps/cli/src/commands/download.command.ts [hunk 2]
```diff
@@ -9,7 +10,10 @@ import { FileResponse } from "../models/response/file.response";
 import { CliUtils } from "../utils";
 
 export abstract class DownloadCommand {
-  constructor(protected cryptoService: CryptoService) {}
+  constructor(
+    protected cryptoService: CryptoService,
+    protected encryptService: EncryptService,
```
[the remark this answers is not in the document]
Comment (quexten) [reply apps/cli/src/commands/download.command.ts]: Removed.

File: libs/tools/export/vault-export/vault-export-core/src/services/individual-vault-export.service.ts [hunk 3]
```diff
@@ -96,7 +101,11 @@ export class IndividualVaultExportService
 
     await Promise.all(promises);
 
-    const encKeyValidation = await this.cryptoService.encrypt(Utils.newGuid());
+    const activeUserId = await firstValueFrom(
```
[the remark this answers is not in the document]
Comment (quexten) [reply libs/tools/export/vault-export/vault-export-core/src/services/individual-vault-export.service.ts:104]: Still using it in line 107: https://github.com/bitwarden/clients/blob/c5e17b6e59d3f57bac520519f7df609f4fd9283d/libs/tools/export/vault-export/vault-export-core/src/services/individual-vault-export.service.ts#L107

**Review by jprusik — APPROVED** [review]
approved for Autofill concerns

File: apps/cli/src/tools/send/commands/receive.command.ts [hunk 4]
```diff
@@ -28,13 +29,14 @@ export class SendReceiveCommand extends DownloadCommand {
 
   constructor(
     private apiService: ApiService,
```
Comment (djsmith85) [inline apps/cli/src/tools/send/commands/receive.command.ts:31]: 🎨 ⛏️  Not scoped for this, but could please also remove the `apiService` as it's not used anywhere.

**Review by djsmith85 — APPROVED** [review]
Approving for Tools

File: apps/cli/src/tools/send/commands/receive.command.ts (hunk 4 above)
Comment (quexten) [reply apps/cli/src/tools/send/commands/receive.command.ts]: Removed

**Review by djsmith85 — APPROVED** [review]

File: libs/common/src/tools/send/models/domain/send.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (justindbaur) [reply libs/common/src/tools/send/models/domain/send.ts:80]: That matches my expectations.

**Review by coroiu — APPROVED** [review]
Only deletions in platform owned files, the best kind of PR 😎 (only reviewed platform code)

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]

**Review by addisonbeck — APPROVED** [review]

**Comment by quexten** [comment]
Resolved merge conflicts. Not sure why rust tests on windows are failing now, but it seems unrelated and also happening on other PRs.

**Review by djsmith85 — DISMISSED** [review]

**Review by coroiu — APPROVED** [review]

**Review by cturnbull-bitwarden — DISMISSED** [review]

**Review by addisonbeck — DISMISSED** [review]

**Review by jprusik — DISMISSED** [review]

**Review by JaredSnider-Bitwarden — DISMISSED** [review]

**Review by gbubemismith — DISMISSED** [review]

**Review by djsmith85 — DISMISSED** [review]

**Review by jprusik — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by jprusik — DISMISSED** [review]

**Review by addisonbeck — APPROVED** [review]

File: libs/common/src/tools/send/models/domain/send.ts [hunk 5]
```diff
@@ -73,9 +73,11 @@ export class Send extends Domain {
     const model = new SendView(this);
 
     const cryptoService = Utils.getContainerService().getCryptoService();
+    const encryptService = Utils.getContainerService().getEncryptService();
 
     try {
-      model.key = await cryptoService.decryptToBytes(this.key, null);
+      const sendKeyEncryptionKey = await cryptoService.getUserKey();
```
Comment (djsmith85) [inline libs/common/src/tools/send/models/domain/send.ts:79]: ❓  @quexten  Should this be `cryptoService.getUserKeyWithLegacySupport` instead? Just thinking about older Sends. Different to ciphers as Sends only have a lifetime of 30 days but want to make sure we are selecting the correct key.

File: libs/common/src/platform/services/cryptography/encrypt.service.implementation.ts [hunk 6]
```diff
@@ -95,6 +95,7 @@ export class EncryptServiceImplementation implements EncryptService {
       const macsEqual = await this.cryptoFunctionService.compareFast(fastParams.mac, computedMac);
       if (!macsEqual) {
         this.logMacFailed("MAC comparison failed. Key or payload has changed.");
+        this.logService.info("key", key.keyB64);
```
Comment (coroiu) [inline libs/common/src/platform/services/cryptography/encrypt.service.implementation.ts:98]: @quexten Are we intentionally writing keys to logs?

File: libs/common/src/tools/send/models/domain/send.ts (hunk 5 above)
Comment (quexten) [reply libs/common/src/tools/send/models/domain/send.ts:79]: cc @jlf0dev We force migrate the legacy users right? So there should be noone with legacy keys for sends?

File: libs/common/src/platform/services/cryptography/encrypt.service.implementation.ts (hunk 6 above)
Comment (quexten) [reply libs/common/src/platform/services/cryptography/encrypt.service.implementation.ts:98]: NO. Good catch!

File: libs/common/src/tools/send/models/domain/send.ts (hunk 5 above)
Comment (jlf0dev) [reply libs/common/src/tools/send/models/domain/send.ts:79]: The only consideration here is making sure that the migration doesn't need this decryption method. I believe it does its own [decryption here](https://github.com/bitwarden/clients/blob/f8fc6269f2bbcaf3713a65ad551c7440baef2c46/libs/common/src/tools/send/services/send.service.ts#L288) so we should be good.

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]

**Review by jprusik — APPROVED** [review]

**Review by addisonbeck — APPROVED** [review]

**Review by coroiu — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**